src/header.rs

```diff
@@ -212,18 +357,24 @@ pub struct TestProps {
     // testing harness and used when generating compilation
     // arguments. (In particular, it propagates to the aux-builds.)
     pub incremental_dir: Option<PathBuf>,
-    // Specifies that a cfail test must actually compile without errors.
-    pub must_compile_successfully: bool,
+    // How far should the test proceed while still passing.
+    pass_mode: Option<PassMode>,
+    // Ignore `--pass` overrides from the command line for this test.
+    ignore_pass: bool,
     // rustdoc will test the output of the `--test` option
     pub check_test_line_numbers_match: bool,
-    // The test must be compiled and run successfully. Only used in UI tests for
-    // now.
-    pub run_pass: bool,
```

Comment:
The removal of the two headers has potential to silently change the result of tests. Do you think it makes sense to deprecate them before removing them?

Reply:
That's a good point. I wonder why they were removed in the original compiletest.

Reply:
IIRC, they were just renamed, not removed. I'm not sure right now which are the correct replacements for each.

Reply:
fwiw, I think it would also be fine to add a changelog that explains the changes =)